client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java

```diff
@@ -0,0 +1,363 @@
+
+    try {
+      planLocalResourceList.add(rssConfFileLocalResource);
+      Field field = DAGProtos.DAGPlan.class.getDeclaredField("localResource_");
+      field.setAccessible(true);
+      field.set(dagPlan, planLocalResourceList);
+      field.setAccessible(false);
+    } catch (Exception e) {
+      LOG.error("submitDAGToAppMaster reflect error", e);
+      throw new TezException(e.getMessage());
+    }
+
+    if (LOG.isDebugEnabled()) {
+      for (DAGProtos.PlanLocalResource localResource : dagPlan.getLocalResourceList()) {
+        LOG.debug("localResource: {}", localResource.toString());
+      }
+    }
+
+    return super.submitDAGToAppMaster(dagPlan, additionalResources);
+  }
+
+  /**
+   * main method
+   * @param args
+   */
+  public static void main(String[] args) {
+    try {
+      // Install the tez class loader, which can be used add new resources
+      TezClassLoader.setupTezClassLoader();
+      Thread.setDefaultUncaughtExceptionHandler(new YarnUncaughtExceptionHandler());
+      final String pid = System.getenv().get("JVM_PID");
+      String containerIdStr =
+              System.getenv(ApplicationConstants.Environment.CONTAINER_ID.name());
+      String appSubmitTimeStr =
+              System.getenv(ApplicationConstants.APP_SUBMIT_TIME_ENV);
+      String clientVersion = System.getenv(TezConstants.TEZ_CLIENT_VERSION_ENV);
+      if (clientVersion == null) {
+        clientVersion = VersionInfo.UNKNOWN;
+      }
+
+      Objects.requireNonNull(appSubmitTimeStr,
+              ApplicationConstants.APP_SUBMIT_TIME_ENV + " is null");
+
+      ContainerId containerId = ConverterUtils.toContainerId(containerIdStr);
+      ApplicationAttemptId applicationAttemptId =
+              containerId.getApplicationAttemptId();
+
+
+      String jobUserName = System
+              .getenv(ApplicationConstants.Environment.USER.name());
+
+      // Command line options
+      Options opts = new Options();
+      opts.addOption(TezConstants.TEZ_SESSION_MODE_CLI_OPTION,
+              false, "Run Tez Application Master in Session mode");
+
+      CommandLine cliParser = new GnuParser().parse(opts, args);
+      boolean sessionModeCliOption = cliParser.hasOption(TezConstants.TEZ_SESSION_MODE_CLI_OPTION);
+
+      LOG.info("Creating DAGAppMaster for "
+              + "applicationId=" + applicationAttemptId.getApplicationId()
+              + ", attemptNum=" + applicationAttemptId.getAttemptId()
+              + ", AMContainerId=" + containerId
+              + ", jvmPid=" + pid
+              + ", userFromEnv=" + jobUserName
+              + ", cliSessionOption=" + sessionModeCliOption
+              + ", pwd=" + System.getenv(ApplicationConstants.Environment.PWD.name())
+              + ", localDirs=" + System.getenv(ApplicationConstants.Environment.LOCAL_DIRS.name())
+              + ", logDirs=" + System.getenv(ApplicationConstants.Environment.LOG_DIRS.name()));
+
+      Configuration conf = new Configuration(new YarnConfiguration());
```

Comment:
Why we need `YarnConfiguration` ?

Reply:
+1. Loading the default yarn conf from classpath ?